src/Compilers/VisualBasic/Test/Semantic/Diagnostics/GetDiagnosticsTests.vb

```diff
@@ -229,8 +229,11 @@ End Namespace
             dict.Add("1", Nothing)
             Dim compilation = CreateCompilation(String.Empty, parseOptions:=New VisualBasicParseOptions().WithPreprocessorSymbols(dict))
 
+            Dim conditionalCompilationConstantNotValidFormat = VBResources.ResourceManager.GetString(
+                NameOf(ERR_ConditionalCompilationConstantNotValid),
+                EnsureEnglishUICulture.PreferredOrNull)
             CompilationUtils.AssertTheseDiagnostics(compilation, <errors>
-BC31030: Conditional compilation constant '1' is not valid: Identifier expected.
+BC31030: <%= String.Format(conditionalCompilationConstantNotValidFormat, VBResources.ERR_ExpectedIdentifier, "1") %>
```

Comment:
><%= String.Format(conditionalCompilationConstantNotValidFormat, VBResources.ERR_ExpectedIdentifier, "1") %> [](start = 9, length = 107)

I don't think this is the right approach. When we are requesting error message for a specific culture, the culture should be propagated to the message arguments as well, but for some reason it is not done here. This looks like a real product issue, not a test issue because I believe our test helper is forcing english culture regardless of default culture on the machine.

Reply:
@MaStr11 

> The diagnostic reported by the compiler is in German. But the assertion logic takes this diagnostic and recreates it enforcing English.

Compiler never reports a diagnostic in any specific language. Language only comes into play when a message is formatted. If Diagnostic.GetMessage is called with specific culture, everything in that message should be in that culture, the template and the arguments. 

> But this does not succeeds for the argument leaving the resulting message as a mixture of German (the argument) and English (the message format string).

This should never happen. The only exception is for arguments that compiler gets from some other APIs as strings. For example, when an exception message is used as an argument for a diagnostic. It doesn't look like any exceptions should be applied for this particular error. So, if you observe that different parts of the message use different language, I would say there is a **real bug** in implementation of Diagnostic.GetMessage API or in compiler, which we should fix.

Reply:
The diagnostic under test is created [here](https://github.com/dotnet/roslyn/blob/master/src/Compilers/VisualBasic/Portable/VisualBasicParseOptions.vb#L255) like so:

`builder.Add(Diagnostic.Create(MessageProvider.Instance, ERRID.ERR_ConditionalCompilationConstantNotValid, VBResources.ERR_ExpectedIdentifier, symbol.Key))`

If I got you you right it should be created like [this one](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/MetadataReader/MetadataHelpers.cs#L926-L927):
```
diagnostics.Add(
    messageProvider.CreateDiagnostic(code, Location.None, 
        new CodeAnalysisResourcesLocalizableErrorArgument(errorArgumentResourceId)));
```
It seems there isn't a `VBResourcesLocalizableErrorArgument` class (at least I couldn't find anything suitable).

I see this options on how to proceed:

* Fix how the argument is passed to the diagnostic. I need some advice how you want this to be done (like *Create a VBResourcesLocalizableErrorArgument class in this location*).
* Revert my changes and let the test fail on non english machines for now. The test needs to be fixed by the compiler team later.

Reply:
There is Microsoft.CodeAnalysis.VisualBasic.LocalizableErrorArgument type. 
Alternatively, for consistency with other places where the same diagnostics is created (that is the approach I would prefer), we should be able to call Diagnostic.Create overload that takes DiagnosticInfo. Like this:
```
Diagnostic.Create(ErrorFactory.ErrorInfo(ERRID.ERR_ConditionalCompilationConstantNotValid,
                                            ErrorFactory.ErrorInfo(ERRID.ERR_ExpectedIdentifier),
                                            symbol.Key))
```